src/rpc/blockchain.cpp

```diff
@@ -491,19 +492,20 @@ static void entryToJSON(const CTxMemPool& pool, UniValue& info, const CTxMemPool
     fees.pushKV("ancestor", ValueFromAmount(e.GetModFeesWithAncestors()));
     fees.pushKV("descendant", ValueFromAmount(e.GetModFeesWithDescendants()));
     info.pushKV("fees", fees);
-
+    if (include_deprecated_fee_fields) {
```

Comment:
Is there a reason you don't do:
```c++
if (IsDeprecatedRPCEnabled("fees")) {
```
here? And not have to change the function header?

Reply:
i was following `ScriptPubKeyToUniv` as an example: https://github.com/bitcoin/bitcoin/blob/42b00a37580a11fbb81a3748d9e834ae8913cdba/src/rpc/blockchain.cpp#L1936-L1939

personally, i agree with this approach more than using the `IsDeprecatedRPCEnabled` function directly as it it's more clear, imo, what's happening and allows for more flexibility. 

for example, lets say we wanted to only allow `getmempoolentry` to return the fields when `-deprecated=fees` is passed but wanted `getrawmempool` to never return the fields. with this setup, we can have one of the functions use the original function signature with `IsDeprecated` passed as the default, whereas we could call `entryToJSON` with `false` passed as the default whenever it's used in `getrawmempool`.

tldr; i think functional overloading is super cool and this seemed a good use case for it :smile:

Reply:
As this `entryToJSON` isn't called anywhere, except for `entryToJSON`, I think it is fine to use the suggestion by @mjdietzx . No strong opinion, though.

Reply:
I agree as well, but no big deal so didn't mention it.

Reply:
seems like its the more preferred approach, happy to switch